### PR TITLE
fix SumkDiscreteFromLattice after TBL changes

### PR DESCRIPTION
**Pull request by the-hampel**
after recent changes in triqs lattice utils Sumk_discrete_from_lattice.py has not been updated. 

* add numpy and fix usage
* exchange `hopping` -> `fourier`
* `SL.dim` is not working anymore, TBL has no property dim anymore, instead use correctly `SL.bl.ndim`

### CODE REVIEW

**Comment by the-hampel** [comment]
todo:
- check if other things are not fixed yet. 
- maybe add a test after correct fix has been implemented.

File: python/triqs/sumk/sumk_discrete_from_lattice.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def __init__(self, lattice, patch = None, n_points = 8, method = "Riemann"):
         self.SL = lattice
         self.patch,self.method = patch,method
         # init the array
-        SumkDiscrete.__init__ (self, dim = self.SL.dim, gf_struct = lattice.OrbitalNames)
+        SumkDiscrete.__init__ (self, dim = self.SL.bl.ndim, gf_struct = lattice.OrbitalNames)
```
Comment (Wentzell) [inline python/triqs/sumk/sumk_discrete_from_lattice.py]:
```suggestion
        SumkDiscrete.__init__ (self, dim = self.SL.ndim, gf_struct = lattice.OrbitalNames)
```

Comment (Wentzell) [reply python/triqs/sumk/sumk_discrete_from_lattice.py]: Thank you for the fix! This class did indeed go unnoticed in the `TBLattice` changes.
I only noticed the simplification above.

A test would certainly be great to have for the `SumkDiscreteFromLattice`!

**Comment by the-hampel** [comment]
I added a very simple test for `sumk_discrete_from_lattice` and also fixed some problems in the class itself.

**Comment by Wentzell** [comment]
Looks good! Thank you @the-hampel

**Pull request merged**